src/services/Credential.js

```diff
@@ -2139,33 +2140,64 @@ class Credential {
 	}
 
 	async ensureDnsValue() {
+		/**
+		 * Define LookupAddress because node doesn't do it
+		 * @typedef dns.LookupAddress
+		 * @property address {String}
+		 * @property family {Number}
+		 */
 
-		const resolveDns = (fqdn) => {
-			const promise = util.promisify(dns.lookup)(fqdn).catch(reason => {
+		/**
+		 * Resolves an fqdn with timeout
+		 * @param fqdn
+		 * @param options
+		 * @returns {Promise<any>}
+		 */
+		function resolveDns(fqdn, options) {
+			const promise = util.promisify(dns.lookup)(fqdn, options).catch(reason => {
 				debug_dns(`Failed to resolve ${fqdn} (${reason})`);
 				throw reason;
 			});
 			return CommonUtils.withTimeout(promise, 3000, new Error('DNS resolution timed out'));
-		};
+		}
+
+		/**
+		 * Validates if the edge can be used
+		 * @param expected_ip {dns.LookupAddress}
+		 * @param real_ip {dns.LookupAddress}
+		 * @param edge_ips { Array.<dns.LookupAddress> | dns.LookupAddress }
+		 */
+		function canUseEdge(expected_ip, real_ip, edge_ips){
+			if(expected_ip.address !== real_ip.address) {
+				return false;
+			}
+			if(Array.isArray(edge_ips)) {
```

Comment:
In which situation `edge_ips` is not an Array? How that can happen?

Reply:
If they only return one IP (in case were we only have 1 edge) it'll not be an array :(

Reply:
documentation says otherwise, so my error, sorry